src/compas/datastructures/mesh/operations/insert.py

```diff
@@ -3,12 +3,69 @@
 from __future__ import division
 
 
-__all__ = []
+__all__ = [
+	'mesh_insert_vertex_on_edge'
+]
+
+
+def mesh_insert_vertex_on_edge(self, u, v, vkey=None):
+	"""Insert an existing vertex on an edge.
+
+	Parameters
+	----------
+	u: hashable
+		The first edge vertex.
+	v: hashable
+		The second edge vertex.
+	vkey: hashable, optional
+		The vertex key to insert.
+		Default is add a new vertex at mid-edge.
+
+	"""

```

Comment:
the behaviour of the function is not entirely clear from the docstring.
also, should it not return the key of the vertex that was inserted?
what happens if the provided vertex key already exists?
...?

Reply:
the description is extended and the key is added as output